### PR TITLE
feat: Examination Result

### LINKED ISSUE #36: [feature] Exam results (semester, overall)
## Requirements

- Expose overall results (cumulative-gpa)
- Expose semester-wise results (sgpa, course metrics like grade, grade point, credits)

## Extra info

On the portal, this information is available at `Examination` > `Examination Result` from the sidebar. For semester-wise results, there's a dropdown that lets you navigate to any semester for which the result has been declared (the default is the last-declared semester).

### PR DESCRIPTION
**Pull request by Blonteractor**
Closes #36, Adds support for fetching the result of examinations, both course wise and overall semester wise result incldung SGPA, CGPA, etc

### CODE REVIEW

**Comment by ditsuke** [comment]
@Blonteractor thanks, the PR looks great! `golangci-lint` is complaining about a non goimports-ed file, could you address it? You should be able to use autofix `golangci-lint run --fix` to fix it.

**Comment by Blonteractor** [comment]
Integeration tests are still failing because the credentials are not set, is that expected? As the secrets are supposed to be set on the repositary itself

**Comment by ditsuke** [comment]
> Integeration tests are still failing because the credentials are not set, is that expected? As the secrets are supposed to be set on the repositary itself

Not fully expected, but its default behavior to to protect GitHub secrets malicious PRs. Workflows need to be edited to enable secret access

File: amizone/models/examination_result.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	"time"
+)
+
+type ExamResultRecord struct {
+	Course CourseRef
+	Result CourseResult
+}
+
+// CourseResult is a model to represent the course wise result in the examinations result page
+type CourseResult struct {
```
Comment (ditsuke) [inline amizone/models/examination_result.go:13]: Perhaps we should nest data for Credit and Grade like
```
{
	score: {Max, Grade, GradePoint}
	Credits: {Acquired, Effective, Points},
	...
}
```

**Review by ditsuke — APPROVED** [review]
I took the liberty of addressing some minor nits and squashing everything down to 5 atomic commits. Looks good to go :rocket:

**Comment by ditsuke** [comment]
@Blonteractor Congrats on your feature contribution to go-amizone. You did a really solid job :100:

**Comment by Blonteractor** [comment]
Thanks man, here's to more 🎉

**Pull request merged**